### PR TITLE
Fixes for Windows

### PR DESCRIPTION
**Pull request by ddd-mtl**
- Less restrictive timestamp comparison as clock is not monotonic
 - Excluded `HostHdk` if target is not wasm32 as HostHdk doesnt link on Windows.

### CODE REVIEW

**Review by neonphog — APPROVED** [review]

File: crates/holochain/src/core/sys_validate.rs [hunk 1]
```diff
@@ -120,7 +120,7 @@ pub async fn check_spam(_header: &Header) -> SysValidationResult<()> {
 
 /// Check previous header timestamp is before this header
 pub fn check_prev_timestamp(header: &Header, prev_header: &Header) -> SysValidationResult<()> {
-    if header.timestamp() > prev_header.timestamp() {
+    if header.timestamp() >= prev_header.timestamp() {
```
Comment (Connoropolous) [inline crates/holochain/src/core/sys_validate.rs:123]: did compiling give you a warn error about this or something?

File: crates/hdk/src/hdk.rs [hunk 2]
```diff
@@ -9,10 +9,10 @@ pub const HDK_NOT_REGISTERED: &str = "HDK not registered";
 /// Every test needs its own mock so each test needs to set it.
 use core::cell::RefCell;
 
-#[cfg(feature = "mock")]
+#[cfg(any(feature = "mock", not(target_arch = "wasm32")))]
```
Comment (Connoropolous) [inline crates/hdk/src/hdk.rs:12]: why do suppose this is necessary?

**Review by Connoropolous — COMMENTED** [review]
Woohoo! Just curiosity questions

File: crates/holochain/src/core/sys_validate.rs (hunk 1 above)
Comment (ddd-mtl) [reply crates/holochain/src/core/sys_validate.rs:123]: I get `CellError(WorkflowError(SourceChainError(InvalidCommit("The previous header\'s timestamp is not before the current header\'s timestamp"))))` most of time without this fix.

File: crates/hdk/src/hdk.rs (hunk 2 above)
Comment (ddd-mtl) [reply crates/hdk/src/hdk.rs:12]: I get `\hdk.dll : fatal error LNK1120: 35 unresolved externals`.

Host functions don't resolve at link by default:
`hdk.hdk.c79tk70i-cgu.8.rcgu.o : error LNK2019: unresolved external symbol __hash_entry referenced in function 
_ZN52_$LT$hdk..hdk..HostHdk$u20$as$u20$hdk..hdk..HdkT$GT$10hash_entry17hb265ccef3ccfb5c3E`

Maybe there is some link config to change but I found it easier to just exclude HostHdk.

**Pull request merged**